R/get_truth.R

```diff
@@ -274,4 +274,41 @@ preprocess_jhu <- function (save_location="./data-truth"){
   colnames(incident_cases)[colnames(incident_cases) == 'inc'] <- 'value'
   incident_cases <- incident_cases[!rowSums(is.na(incident_cases[c("location_name","value")])), ]
   readr::write_csv(incident_cases, path = file.path(save_location,"truth-Incident Cases.csv"))
+}
+
+
+#' Preprocess raw hospitalization data into Cumulative/Incident hospitalizations and write to CSVs
+#' 
+#' @param save_location character specifying the location of to save raw truth data.
+#' Default to "./data-truth"
+#' 
+#' @return data frame of cumulative and incident hospitalization data
+#' 
+#' @export
+preprocess_hospitalization <- function (save_location="./data-truth"){
+  # Location name
+  location_names <- covidData::fips_codes[, c("location", "location_name")]
+  
+  # Spatial resolutions
+  spatial_resolutions <- c("national", "state")
+  
+  # Combine to get cases data
+  hospitalization_dframes <- covidData::load_healthdata_data(spatial_resolution = spatial_resolutions,
+                                                             temporal_resolution = "daily",
+                                                             measure = "hospitalizations")
+  hospitalization_dframes <- dplyr::left_join(hospitalization_dframes, location_names, by = "location")
+
+  # Get cumulative hospitalizations
+  cumulative_hosp <- hospitalization_dframes[, c("date", "location", "location_name", "cum")]
+  colnames(cumulative_hosp)[colnames(cumulative_hosp) == 'cum'] <- 'value'
+  cumulative_hosp <- cumulative_hosp[!rowSums(is.na(cumulative_hosp[c("location_name","value")])), ]
```

Comment:
Can we add a comment here and on line 310 below, or in the function definition, explaining what this filter on NA values is doing?  What are the cases where the location name is NA, or where the value is NA?

Reply:
The original python script that creates the standard truth csv files for cumulative/incident cases/deaths were filtering out these rows [here](https://github.com/reichlab/covid19-forecast-hub/blob/master/data-truth/get-truth-data.py#L40), so I was thinking for hospitalization truth I should do the exact same steps, hence the filtering out NA values was included. I think because these truths are used for visualization, and missing values will cause errors I believe, so rows with these missing values are filtered.

Reply:
Since we don't use it for visualization yet, maybe we don't need this filtering out?